src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs

```diff
@@ -3665,7 +3665,89 @@
+    ("SveGatherVectorFirstFaultingIndices.template",   new Dictionary<string, string> { ["TestName"] = "Sve_GatherVectorInt16SignExtendFirstFaulting_Indices_ulong_long",                                                    ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "GatherVectorInt16SignExtendFirstFaulting",                                          ["RetVectorType"] = "Vector",    ["RetBaseType"] = "UInt64",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt64",                                  ["Op2BaseType"] = "Int16",  ["Op3VectorType"] = "Vector",    ["Op3BaseType"] = "Int64",  ["ExtendedElementType"] = "Int16",    ["GetFfrType"] = "UInt64",      ["LargestVectorSize"] = "64", ["NextValueOp1"] = "Helpers.getMaskUInt64()",      ["NextValueOp2"] = "TestLibrary.Generator.GetInt16()",                     ["NextValueOp3"] = "TestLibrary.Generator.GetInt64()"}),
+    ("SveGatherVectorFirstFaultingIndices.template",   new Dictionary<string, string> { ["TestName"] = "Sve_GatherVectorInt16SignExtendFirstFaulting_Indices_ulong_ulong",                                                   ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "GatherVectorInt16SignExtendFirstFaulting",                                          ["RetVectorType"] = "Vector",    ["RetBaseType"] = "UInt64",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt64",                                  ["Op2BaseType"] = "Int16",  ["Op3VectorType"] = "Vector",    ["Op3BaseType"] = "UInt64", ["ExtendedElementType"] = "Int16",    ["GetFfrType"] = "UInt64",      ["LargestVectorSize"] = "64", ["NextValueOp1"] = "Helpers.getMaskUInt64()",      ["NextValueOp2"] = "TestLibrary.Generator.GetInt16()",                     ["NextValueOp3"] = "TestLibrary.Generator.GetUInt64()"}),
+    ("SveGatherVectorFirstFaultingIndices.template",   new Dictionary<string, string> { ["TestName"] = "Sve_GatherVectorInt16WithByteOffsetsSignExtendFirstFaulting_offsets_int_int",                                        ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "GatherVectorInt16WithByteOffsetsSignExtendFirstFaulting",                           ["RetVectorType"] = "Vector",    ["RetBaseType"] = "Int32",   ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Int32",                                   ["Op2BaseType"] = "Int16",  ["Op3VectorType"] = "Vector",    ["Op3BaseType"] = "Int32",  ["ExtendedElementType"] = "Int16",    ["GetFfrType"] = "UInt32",      ["LargestVectorSize"] = "64", ["NextValueOp1"] = "Helpers.getMaskInt32()",       ["NextValueOp2"] = "TestLibrary.Generator.GetInt16()",                     ["NextValueOp3"] = "TestLibrary.Generator.GetInt32()"}),
```

Comment:
For the `WithByteOffsets` APIs, should we be using `SveGatherVectorFirstFaultingIndices.template`, or do we need a new template that works specifically with byte offsets?

Reply:
You're right that the names maybe a little misleading. The `*FirstFaultingIndices.template` take a pointer in op2 and indices in op3 while `*FirstFaultingBases.template` take a vector of addresses.
Would you suggest to change their names to something different?

Reply:
I think the naming is fine. My concern is this template is treating `op3` like a Vector of indices, instead of a Vector of byte offsets. For example, suppose the data source is an array of `Int64`. In this template's implementation of `RunBasicScenario_LoadFirstFaulting`, we set the last element of `op3` to `outsideBoundsIndex` to force a fault. `outsideBoundsIndex` is equal to `(Vector<{Int64}>.Count - 1)`, so with a 128-bit width, it's `outsideBoundsIndex = 2 - 1 = 1`. An index of 1 would be out-of-bounds, but a byte offset of 1 shouldn't be, so I'd expect this to not fault. To cause a fault, I think we'd need to set the last element of `op3` to `outsideBoundsIndex * sizeof(Int64)`, or 8 bytes.

I think you can reuse the new template I'm adding in #106199 to test the byte offset APIs.

Reply:
Agreed, makes sense. I'll let your patch merge and then update this PR.